### PR TITLE
check git version in GitRepository#isWorking()

### LINKED ISSUE #2664: GitRepository isWorking should check git version
As we are using features from newer git version (date format), the repository should report if it's working via the `isWorking()` method. Currently just return code is checked.

```
    @Override
    public boolean isWorking() {
        if (working == null) {
            ensureCommand(CMD_PROPERTY_KEY, CMD_FALLBACK);
            working = checkCmd(RepoCommand, "--help");
        }
        return working;
    }
```

### PR DESCRIPTION
**Pull request by tulinkry**
fixes #2664

### CODE REVIEW

File: opengrok-indexer/src/main/java/org/opengrok/indexer/util/Version.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+ * file and include the License file at LICENSE.txt.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+/*
+ * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
```
Comment (vladak) [inline opengrok-indexer/src/main/java/org/opengrok/indexer/util/Version.java]: I think you should have @tulinkry there

File: opengrok-indexer/src/test/java/org/opengrok/indexer/util/VersionTest.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
+package org.opengrok.indexer.util;
```
Comment (vladak) [inline opengrok-indexer/src/test/java/org/opengrok/indexer/util/VersionTest.java]: missing license

File: opengrok-indexer/src/main/java/org/opengrok/indexer/util/Version.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
```
Comment (vladak) [inline opengrok-indexer/src/main/java/org/opengrok/indexer/util/Version.java]: I mean you do not have to include Oracle copyright at all if all of the code comes from elsewhere.

Comment (tulinkry) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/util/Version.java]: I don't mind about putting oracle license there.

**Pull request merged**